app/components/candidate_interface/degree_grade_component.html.erb

```diff
@@ -1,24 +1,50 @@
 <% content_for :title, title_with_error_prefix(legend_helper, @model.errors.any?) %>
-<%= form_with model: @model, url: candidate_interface_degree_grade_path do |form| %>
+<%= form_with model: @model, url: candidate_interface_degree_grade_path, scope: :candidate_interface_degree_form do |form| %>
   <%= form.govuk_error_summary %>
-  <%= form.govuk_radio_buttons_fieldset :grade, legend: { text: legend_helper, tag: 'h1', size: 'l' } do %>
-    <% if @model.uk? && specific_grade_options? %>
-      <% grades.each_with_index do |grade, index| %>
+  <%= form.govuk_radio_buttons_fieldset(
+    :grade,
+    legend: { text: legend_helper, tag: 'h1', size: 'l' },
+    hint:,
+  ) do %>
+    <% if show_uk_grades? %>
+      <% uk_grades.each_with_index do |grade, index| %>
         <%= form.govuk_radio_button :grade, grade, label: { text: grade }, link_errors: index.zero? do %>
           <% if grade == 'Other' %>
               <%= render DfE::Autocomplete::View.new(
                 form,
                 attribute_name: :other_grade,
                 form_field: form.govuk_select(
                   :other_grade,
-                  options_for_select(dfe_autocomplete_options(@model.other_grades), form.object.other_grade.to_s),
+                  options_for_select(dfe_autocomplete_options(@model.other_grades), @model.other_grade.to_s),
                   label: { text: t('application_form.degree.grade.other.conditional.label') },
                   width: 10,
                 ),
               ) %>
           <% end %>
         <% end %>
       <% end %>
+    <% elsif show_compatible_country_grades? %>
```

Comment:
this is new functionality. we show different grades for degrees with uk-compatible degrees.